website/src/content/blog/2025-03-31-typespec-1-0-release/typespec_1_0_release.md

```diff
@@ -0,0 +1,215 @@
+  - [@typespec/http-client-java](https://www.npmjs.com/package/@typespec/http-client-java)
+  - [@typespec/http-client-python](https://www.npmjs.com/package/@typespec/http-client-python)
+  - [@typespec/http-server-csharp](https://www.npmjs.com/package/@typespec/http-server-csharp)
+  - [@typespec/http-server-js](https://www.npmjs.com/package/@typespec/http-server-js)
+
+## TypeSpec in action
+
+Let's see how TypeSpec works with a simple Todo API example. First, you define your API using TypeSpec's concise syntax:
+
+```tsp title=main.tsp tryit="{"emit": ["@typespec/openapi3"]}"
+import "@typespec/http";
+import "@typespec/rest";
+using Http;
+using Rest;
+
+@route("/todoitems")
+interface TodoItems {
+  @get getTodoItems(): TodoItem[];
+  @post createTodoItem(@body body: CreateTodoItem): Http.CreatedResponse & TodoItem;
+}
+
+model TodoItem {
+  @visibility(Lifecycle.Read)
+  id: string;
+
+  content: string;
+  dueDate: utcDateTime;
+  isCompleted: boolean;
+  labels?: string[];
+}
+
+model CreateTodoItem {
+  content: string;
+  labels?: string[];
+}
+```
+
+From this single definition, TypeSpec can generate multiple artifacts that keep your API implementation consistent across languages and platforms. One of the most powerful capabilities is the ability to generate both client and server code directly from your TypeSpec definition.
+
+## Code generation (Preview)
+
+One of the exciting new features in TypeSpec 1.0-RC is the ability to generate both server-side and client-side code. This feature is currently in preview, and we're actively seeking feedback to refine it.
+
+![TypeSpec Workflow Diagram](./workflow-diagram-full.png)
+
+### Server-side code
+
+With the server-side code generators, TypeSpec can produce controller and model code from your API definition that serves as a foundation for your implementation:
+
+```csharp
+[HttpGet]
+[Route("/todoitems")]
+[ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(TodoItem[]))]
+public virtual async Task<IActionResult> GetTodoItems()
+{
+    var result = await TodoItemsOperationsImpl.GetTodoItemsAsync();
+    return Ok(result);
+}
+```
+
+### Client libraries
+
+TypeSpec can also generate client libraries in multiple languages, providing a consistent, type-safe interface to your API:
+
+```javascript
+const client = new TodoServiceClient();
+const todo = await client.createTodoItem({
+  content: "Buy groceries",
+  dueDate: new Date("2025-03-31"),
+  isCompleted: false,
+});
+console.log(todo);
+```
+
+### Current limitations
+
+The code generation features are currently in preview and have a few technical limitations to be aware of:
+
+- Limited support for authentication mechanisms
+- Complex data transformations may require manual coding
+- Not all HTTP features are fully supported
+- Documentation for these features is still being developed
+
+We're actively working to address these limitations and we welcome your feedback.
```

Comment:
I didn't think we'll ever support all http features. 
This is part of what scares me: how much resources will we put into features that azure itself doesn't need? I'd be very cautious of making any promises.